### ISSUE TITLE
DELL XPS 17 - 9730H Intel® Smart Sound Technology for MIPI SoundWire® Audio

### ISSUE THREAD

**Issue by KarthikBala2003** [thesofproject/linux#4380]
[Bug 217461 ](https://bugzilla.kernel.org/show_bug.cgi?id=217461)

Ubuntu (Lobster) 23.04 not recognizing Intel® Smart Sound Technology for MIPI SoundWire® Audio.
gnome-control-center showing dummy-output. 
I tried with the latest mainline kernel 6.4.00-RC2, still does sound not working.

Here is the output of sudo inxi -SMA
Audio:
  Device-1: Intel driver: sof-audio-pci-intel-tgl
  Sound API: ALSA v: k6.4.0-060400rc2-generic running: yes
  Sound Server-1: PipeWire v: 0.3.65 running: yes
[alsa-info.txt](https://github.com/thesofproject/linux/files/11568265/alsa-info.txt)
[Audio Device.txt](https://github.com/thesofproject/linux/files/11568266/Audio.Device.txt)
[dmesg.txt](https://github.com/thesofproject/linux/files/11568267/dmesg.txt)
![Sound-Dummy-Output](https://github.com/thesofproject/linux/assets/12576200/cf42e865-8ad5-49f3-8447-c5d86fc96ff5)
![Ubuntu-Info](https://github.com/thesofproject/linux/assets/12576200/22d6aab9-deba-47d7-a75c-6366b1ce14d1)

I reinstalled but the issue still persists on my laptop.
I can connect any external USB audio device and external speakers works fine. I can see both digital and analog Output Devices listed, I tried with Creative Pebble Pro.

This issue only occurs integrated sound card on my XPS 17 13th Gen Intel i9-13900H. See the attached device Audio device info captured from Windows 11 OS.
Audio working fine with Windows 11 OS.

fwupdmgr reinstall
Selected device: TPS6598X PD#0
╔══════════════════════════════════════════════════════════════════════════════╗
║ Reinstall TPS6598X PD#0 to F907.14.13?                                       ║
╠══════════════════════════════════════════════════════════════════════════════╣
║ This stable release fixes the following bug:                                 ║
║                                                                              ║
║ • Improve USB-C monitor compatibility                                        ║
║                                                                              ║
║ TPS6598X PD#0 and all connected devices may not be usable while updating.    ║
╚══════════════════════════════════════════════════════════════════════════════╝
Perform operation? [Y|n]: Y
Waiting…                 [***************************************] Less than one minute remaining…
Successfully installed firmware

**iasl -d dsdt.dat**
[dsdt.txt](https://github.com/thesofproject/linux/files/11568307/dsdt.txt)

Intel ACPI Component Architecture
ASL+ Optimizing Compiler/Disassembler version 20200925
Copyright (c) 2000 - 2020 Intel Corporation
[dsdt.zip](https://github.com/thesofproject/linux/files/11568329/dsdt.zip)

**Comment by plbossart** [comment]
Bad news @KarthikBala2003, this Dell `System SKU:        0BDA` is currently not supported in the Linux kernel.
This must be a Windows-only device for which Dell does not provide a Linux OEM image?

This looks like a pretty standard topology if I look at the DSDT:

Name (_ADR, 0x000030025D071101) -> headset codec on link0
Name (_ADR, 0x000131025D131601)  -> amplifier on link1 and link2
Name (_ADR, 0x000230025D131601)
Name (_ADR, 0x000330025D071401)  -> microphone on link3

The PCI device is [8086:51ca] so that's

````
{ PCI_DEVICE(0x8086, 0x51ca), /* RPL-P */
````

So what's missing is the ACPI tables for this in the kernel, along with the DMI quirk for the headset.

**Comment by plbossart** [comment]
@KarthikBala2003 please try this: https://github.com/thesofproject/linux/pull/4381

If you are not familiar with kernel compilation, see https://thesofproject.github.io/latest/getting_started/setup_linux/install_locally.html

**Comment by KarthikBala2003** [comment]
I really appreciated your immediate response and troubleshooting of the issue. 
I'm not very familiar to build a custom kernel from source. Can someone help me with the steps involved to build the kernel to add a table for RPL Dell SKU OBDA. 

I'm very much interested in learning this new process, and how can I pull the changes and compile locally to make sure the fix working as expected. 

Is there any possibility new kernel will support it?

**Comment by plbossart** [comment]
@KarthikBala2003 I provided you with the steps, I can't help further.

If this fix works, it'll be merged in the Linux kernel 6.5 at best, and probably back-ported to linux stable versions. We're talking about at least 3 months before you see a solution, and you will still need to unzip and add this file 
[sof-rpl-rt711-l0-rt1316-l12-rt714-l3.tplg.gz](https://github.com/thesofproject/linux/files/11569351/sof-rpl-rt711-l0-rt1316-l12-rt714-l3.tplg.gz) in /lib/firmware/intel/sof-tplg/

**Comment by KarthikBala2003** [comment]
when I execute the below command, it failed with below error
make -j$(nproc --all)

CC [M]  drivers/gpu/drm/i915/i915_vgpu.o
  LD [M]  drivers/gpu/drm/i915/i915.o
  AR      drivers/gpu/built-in.a
  AR      drivers/built-in.a
make: *** [Makefile:2026: .] Error 2

**Comment by KarthikBala2003** [comment]
Can I run through this step again for the custom kernel build? What step I need to perform, I downloaded the zip file and did run the steps provide in the link but failed executing the command for kernel compilation **_make -j$(nproc --all)_**
[Install the Kernel Locally](https://thesofproject.github.io/latest/getting_started/setup_linux/install_locally.html)

**Comment by plbossart** [comment]
Try to install the pre-build linux-image we used in our CI: 
[linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb.gz](https://github.com/thesofproject/linux/files/11576849/linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb.gz)

unzip the file and install with "sudo dpkg -i linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb"
Disclaimer: use at your own risk, etc. I am just providing this binary as a means to help test.

**Comment by KarthikBala2003** [comment]
When executes as root user, getting this error.

sudo dpkg -i linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb

dpkg: error: archive 'linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb' is not a regular file

**Comment by KarthikBala2003** [comment]
I tried with -P flag, here is the output

sudo dpkg -P linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64
dpkg: warning: ignoring request to remove linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64 which isn't installed

**Comment by plbossart** [comment]
Works for me:

````
sudo dpkg -i ~/Downloads/linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb 
Selecting previously unselected package linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f.
(Reading database ... 300022 files and directories currently installed.)
Preparing to unpack .../linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb ...
Unpacking linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f (6.4.0-rc1-812ee1d6) ...
Setting up linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f (6.4.0-rc1-812ee1d6) ...
 * dkms: running auto installation service for kernel 6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f                                                                                                           [ OK ] 
update-initramfs: Generating /boot/initrd.img-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f
I: The initramfs will attempt to resume from /dev/sda2
I: (UUID=37bc8a34-6286-4ac9-a98f-c71b7e5954ce)
I: Set the RESUME variable to override this.
Sourcing file `/etc/default/grub'
Sourcing file `/etc/default/grub.d/init-select.cfg'
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f
Found initrd image: /boot/initrd.img-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f
````

**Comment by KarthikBala2003** [comment]
I will move this package to Download Folder and try again

**Comment by KarthikBala2003** [comment]
Same erro, will reboot and test it again

dpkg: error: archive '/home/user/Downloads/linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb' is not a regular file

**Comment by KarthikBala2003** [comment]
For some reason, I can't run the deb package, same error  
dpkg: error: archive 'linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb' is not a regular file.

I tried from the Downloads folder as a root user with no luck.  I'm using Ubuntu 23.04. Is there any other option to execute this package? Can we repackage it?

**Comment by KarthikBala2003** [comment]
Is it possible different compression

**Comment by KarthikBala2003** [comment]
Is it possible to use a different compression form to support Ubuntu 23.04 on 6.4.0-060400rc3-generic?

**Comment by plbossart** [comment]
make sure you decompress with gunzip and verify the hash:

md5sum linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb
418bb0e7503a9079ad29e213d3bc139a  linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb

**Comment by KarthikBala2003** [comment]
Good Catch. I didn't use gunzip rather tried with the default archive extract.| I successfully installed it. Will reboot and try the sound and share the results

**Comment by KarthikBala2003** [comment]
No luck,  I installed and rebooted. still, the fix is not working.  Attached dsdt details for analysis. neofetch and alsa-info.txt details are attached as well. 


inxi -SMA
System:
  Host: XPS17-9730 Kernel: 6.4.0-060400rc3-generic arch: x86_64 bits: 64
[dsdt.zip](https://github.com/thesofproject/linux/files/11579310/dsdt.zip)

    Desktop: GNOME v: 44.0 Distro: Ubuntu 23.04 (Lunar Lobster)
Machine:
  Type: Laptop System: Dell product: XPS 17 9730 v: N/A
    serial: <superuser required>
  Mobo: Dell model: 01Y5VN v: A00 serial: <superuser required> UEFI: Dell
    v: 1.2.2 date: 03/23/2023
Audio:
  Device-1: Intel driver: sof-audio-pci-intel-tgl
  Sound API: ALSA v: k6.4.0-060400rc3-generic running: yes
  Sound Server-1: PipeWire v: 0.3.65 running: yes
[alsa-info.txt](https://github.com/thesofproject/linux/files/11579630/alsa-info.txt)
![Screenshot from 2023-05-26 15-02-21](https://github.com/thesofproject/linux/assets/12576200/adc69bf1-80f4-4c5c-933f-c9f17821efbc)

**Comment by plbossart** [comment]
please add this file 
[sof-dyndbg.conf.txt](https://github.com/thesofproject/linux/files/11579664/sof-dyndbg.conf.txt) as /etc/modprobe.d/sof-dyndbg.conf, reboot and attach the dmesg log.

**Comment by KarthikBala2003** [comment]
I didn't find this file.

ll /etc/modprobe.d/
total 52
drwxr-xr-x 1 root root  554 May 25 22:05 ./
drwxr-xr-x 1 root root 4546 May 26 12:43 ../
-rw-r--r-- 1 root root 2507 Feb 22  2021 alsa-base.conf
-rw-r--r-- 1 root root  154 Aug 23  2022 amd64-microcode-blacklist.conf
-rw-r--r-- 1 root root  325 Feb 15 04:04 blacklist-ath_pci.conf
-rw-r--r-- 1 root root 1518 Feb 15 04:04 blacklist.conf
-rw-r--r-- 1 root root  210 Feb 15 04:04 blacklist-firewire.conf
-rw-r--r-- 1 root root  677 Feb 15 04:04 blacklist-framebuffer.conf
-rw-r--r-- 1 root root  156 Feb 22  2021 blacklist-modem.conf
lrwxrwxrwx 1 root root   41 Apr 18 15:34 blacklist-oss.conf -> /lib/linux-sound-base/noOSS.modprobe.conf
-rw-r--r-- 1 root root  583 Feb 15 04:04 blacklist-rare-network.conf
-rw-r--r-- 1 root root  127 Jan 31 05:24 dkms.conf
-rw-r--r-- 1 root root  154 Feb 19 18:02 intel-microcode-blacklist.conf
-rw-r--r-- 1 root root  347 Feb 15 04:04 iwlwifi.conf
-rw-r--r-- 1 root root  117 Mar 27 06:29 nvidia-graphics-drivers-kms.conf

**Comment by KarthikBala2003** [comment]
Sorry, I misunderstood.  
I copied the file as sof-dyndbg.conf, into etc/modprobe.d/ and rebooted. After reboot I captured dmesg. 

ll /etc/modprobe.d/
total 56
drwxr-xr-x 1 root root  584 May 26 17:12 ./
drwxr-xr-x 1 root root 4546 May 26 12:43 ../
-rw-r--r-- 1 root root 2507 Feb 22  2021 alsa-base.conf
-rw-r--r-- 1 root root  154 Aug 23  2022 amd64-microcode-blacklist.conf
-rw-r--r-- 1 root root  325 Feb 15 04:04 blacklist-ath_pci.conf
-rw-r--r-- 1 root root 1518 Feb 15 04:04 blacklist.conf
-rw-r--r-- 1 root root  210 Feb 15 04:04 blacklist-firewire.conf
-rw-r--r-- 1 root root  677 Feb 15 04:04 blacklist-framebuffer.conf
-rw-r--r-- 1 root root  156 Feb 22  2021 blacklist-modem.conf
lrwxrwxrwx 1 root root   41 Apr 18 15:34 blacklist-oss.conf -> /lib/linux-sound-base/noOSS.modprobe.conf
-rw-r--r-- 1 root root  583 Feb 15 04:04 blacklist-rare-network.conf
-rw-r--r-- 1 root root  127 Jan 31 05:24 dkms.conf
-rw-r--r-- 1 root root  154 Feb 19 18:02 intel-microcode-blacklist.conf
-rw-r--r-- 1 root root  347 Feb 15 04:04 iwlwifi.conf
-rw-r--r-- 1 root root  117 Mar 27 06:29 nvidia-graphics-drivers-kms.conf
-rw-r--r-- 1 root root 2739 May 26 17:12 **_sof-dyndbg.conf_**


[dmesg.txt](https://github.com/thesofproject/linux/files/11580262/dmesg.txt)

**Comment by KarthikBala2003** [comment]
The audio issue still persists, even I installed patch kernel module.
Please let me know if you find additional details from the latest attached dmesg

**Comment by plbossart** [comment]
There's still something preventing the SoundWire initialization from happening, this sort of messages should not happen

````
[   19.006842] soundwire_intel:intel_link_process_wakeen_event: soundwire_intel soundwire_intel.link.0: SoundWire master 0 is disabled or not-started, ignoring
[   19.006844] soundwire_intel:intel_link_process_wakeen_event: soundwire_intel soundwire_intel.link.1: SoundWire master 1 is disabled or not-started, ignoring
[   19.006845] soundwire_intel:intel_link_process_wakeen_event: soundwire_intel soundwire_intel.link.2: SoundWire master 2 is disabled or not-started, ignoring
[   19.006847] soundwire_intel:intel_link_process_wakeen_event: soundwire_intel soundwire_intel.link.3: SoundWire master 3 is disabled or not-started, ignoring
````

We've had a similar issue recently with an ACPI quirk not being detected correctly. Let me think about this a bit more @KarthikBala2003

**Comment by plbossart** [comment]
@KarthikBala2003 Are you sure you rebooted in the changed kernel?

The dmesg should start with

````
[    0.000000] kernel: Linux version 6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f (jenkins@sofbld13.ostc.intel.com) (gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #812ee1d6 SMP PREEMPT_DYNAMIC Thu May 25 13:56:37 PDT 2023
````

What you have does not match this.

````
[    0.000000] Linux version 6.4.0-060400rc3-generic (kernel@kathleen) (x86_64-linux-gnu-gcc-12 (Ubuntu 12.2.0-17ubuntu1) 12.2.0, GNU ld (GNU Binutils for Ubuntu) 2.40) #202305212230 SMP PREEMPT_DYNAMIC Sun May 21 22:36:19 UTC 2023
[    0.000000] Command line: BOOT_IMAGE=/vmlinuz-6.4.0-060400rc3-generic root=UUID=ab78c21a-c3ff-45ff-9b6a-203d8433f1d2 ro persistent
````

Wrong version, different tools and date that is clearly BEFORE you submitted the issue...

**Comment by KarthikBala2003** [comment]
That is correct, I have updated the kernel version to RC3. Is there any problem with this version. Do I need to downgrade the version?
But I followed the process and rebooted, after that I didn't do any upgrade

**Comment by plbossart** [comment]
Well if we give you a version to test and you try something else, we are not going to make a lot of progress... You need to select the test version in the grub menu. Rebooting is not enough if you don't select the relevant version to test.

**Comment by KarthikBala2003** [comment]
I selected the rest release version kernel: Linux version 6.4.0-rc1 and attached the results here for your analysis. 

inxi -SMA
System:
  Host: XPS17-9730 Kernel: 6.4.0-060400rc1-generic arch: x86_64 bits: 64
    Desktop: GNOME v: 44.0 Distro: Ubuntu 23.04 (Lunar Lobster)
Machine:
  Type: Laptop System: Dell product: XPS 17 9730 v: N/A
    serial: <superuser required>
  Mobo: Dell model: 01Y5VN v: A00 serial: <superuser required> UEFI: Dell
    v: 1.2.2 date: 03/23/2023
Audio:
  Device-1: Intel driver: sof-audio-pci-intel-tgl
  Sound API: ALSA v: k6.4.0-060400rc1-generic running: yes
  Sound Server-1: PipeWire v: 0.3.65 running: yes

ll /etc/modprobe.d/
total 60
drwxr-xr-x 1 root root  604 May 27 16:17 ./
drwxr-xr-x 1 root root 4770 May 30 20:55 ../
-rw-r--r-- 1 root root 2507 Feb 22  2021 alsa-base.conf
-rw-r--r-- 1 root root  154 Aug 23  2022 amd64-microcode-blacklist.conf
-rw-r--r-- 1 root root  325 Feb 15 04:04 blacklist-ath_pci.conf
-rw-r--r-- 1 root root 1518 Feb 15 04:04 blacklist.conf
-rw-r--r-- 1 root root  210 Feb 15 04:04 blacklist-firewire.conf
-rw-r--r-- 1 root root  677 Feb 15 04:04 blacklist-framebuffer.conf
-rw-r--r-- 1 root root  156 Feb 22  2021 blacklist-modem.conf
lrwxrwxrwx 1 root root   41 Apr 18 15:34 blacklist-oss.conf -> /lib/linux-sound-base/noOSS.modprobe.conf
-rw-r--r-- 1 root root  583 Feb 15 04:04 blacklist-rare-network.conf
-rw-r--r-- 1 root root  127 Jan 31 05:24 dkms.conf
-rw-r--r-- 1 root root  154 Feb 19 18:02 intel-microcode-blacklist.conf
-rw-r--r-- 1 root root  347 Feb 15 04:04 iwlw
[dsdt.zip](https://github.com/thesofproject/linux/files/11607929/dsdt.zip)
![XPS17-Neofetch-Kernel-6 4-RC1](https://github.com/thesofproject/linux/assets/12576200/dbe42d55-0619-4717-bfa1-d2a2de140f86)
[alsa-info.txt](https://github.com/thesofproject/linux/files/11607935/alsa-info.txt)
[dmesg.txt](https://github.com/thesofproject/linux/files/11607936/dmesg.txt)
![XPS17-Neofetch-Kernel-6 4-RC1](https://github.com/thesofproject/linux/assets/12576200/d16b14cb-9d76-45d6-856a-07e189810a25)
ifi.conf
-rw-r--r-- 1 root root  379 Jan  7  2022 mdadm.conf
-rw-r--r-- 1 root root  117 Mar 27 06:29 nvidia-graphics-drivers-kms.conf
-rw-r--r-- 1 root root 2739 May 30 20:53 sof-dyndbg.conf

**Comment by KarthikBala2003** [comment]
Could you please let me know if I missed anything here, I still have no audio enabled.

**Comment by plbossart** [comment]
I don't think I can help further. I provided you with a debian package and you are not trying to install it. If you use an 'generic' kernel nothing will happen magically.

**Comment by KarthikBala2003** [comment]
I understand your concern, I already installed the package you sent me and added sof-dyndbg.conf, rebooted.

**Comment by KarthikBala2003** [comment]
My apology. since I'm doing this first time and trying to understand the process and determine the cause of this issue.
This is very critical for me. Please help me with this issue. 

I tried both our test compiled kernel and 6.4-RC1 both not working as expected.  I have attached the log as well.

I also noticed that the Bluetooth module also greyed out if I use this test kernel.
[alsa-info.txt](https://github.com/thesofproject/linux/files/11614987/alsa-info.txt)
[dmesg.txt](https://github.com/thesofproject/linux/files/11614988/dmesg.txt)
[dsdt.zip](https://github.com/thesofproject/linux/files/11614989/dsdt.zip)


uname -r
**6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f**

sudo dpkg -i linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb

(Reading database ... 327836 files and directories currently installed.)
Preparing to unpack linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f_6.4.0-rc1-812ee1d6_amd64.deb ...
Unpacking linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f (6.4.0-rc1-812ee1d6) over (6.4.0-rc1-812ee1d6) ...
Setting up linux-image-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f (6.4.0-rc1-812ee1d6) ...
 * dkms: running auto installation service for kernel 6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f                       [ OK ] 
update-initramfs: Generating /boot/initrd.img-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f
I: The initramfs will attempt to resume from /dev/nvme0n1p4
I: (UUID=862b6240-4a21-4c34-8015-e92a0bb310ae)
I: Set the RESUME variable to override this.
Sourcing file `/etc/default/grub'
Sourcing file `/etc/default/grub.d/init-select.cfg'
Generating grub configuration file ...
Found theme: /usr/share/grub/themes/Tela/theme.txt
Found linux image: /boot/vmlinuz-6.4.0-060400rc4-generic
Found initrd image: /boot/initrd.img-6.4.0-060400rc4-generic
Found linux image: /boot/vmlinuz-6.4.0-060400rc3-generic
Found initrd image: /boot/initrd.img-6.4.0-060400rc3-generic
Found linux image: /boot/vmlinuz-6.4.0-060400rc1-generic
Found initrd image: /boot/initrd.img-6.4.0-060400rc1-generic
Found linux image: /boot/vmlinuz-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f
Found initrd image: /boot/initrd.img-6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f
Found linux image: /boot/vmlinuz-6.2.0-20-generic
Found initrd image: /boot/initrd.img-6.2.0-20-generic
Found memtest86+ 64bit EFI image: /memtest86+x64.efi
Warning: os-prober will be executed to detect other bootable partitions.
Its output will be used to detect bootable binaries on them and create new boot entries.
Found Windows Boot Manager on /dev/nvme1n1p1@/efi/Microsoft/Boot/bootmgfw.efi
Adding boot menu entry for UEFI Firmware Settings ...
done

sudo cp sof-dyndbg.conf.txt /etc/modprobe.d/sof-dyndbg.conf

ll /etc/modprobe.d
total 60
drwxr-xr-x 1 root root  604 May 27 16:17 ./
drwxr-xr-x 1 root root 4770 May 31 07:56 ../
-rw-r--r-- 1 root root 2507 Feb 22  2021 alsa-base.conf
-rw-r--r-- 1 root root  154 Aug 23  2022 amd64-microcode-blacklist.conf
-rw-r--r-- 1 root root  325 Feb 15 04:04 blacklist-ath_pci.conf
-rw-r--r-- 1 root root 1518 Feb 15 04:04 blacklist.conf
-rw-r--r-- 1 root root  210 Feb 15 04:04 blacklist-firewire.conf
-rw-r--r-- 1 root root  677 Feb 15 04:04 blacklist-framebuffer.conf
-rw-r--r-- 1 root root  156 Feb 22  2021 blacklist-modem.conf
lrwxrwxrwx 1 root root   41 Apr 18 15:34 blacklist-oss.conf -> /lib/linux-sound-base/noOSS.modprobe.conf
-rw-r--r-- 1 root root  583 Feb 15 04:04 blacklist-rare-network.conf
-rw-r--r-- 1 root root  127 Jan 31 05:24 dkms.conf
-rw-r--r-- 1 root root  154 Feb 19 18:02 intel-microcode-blacklist.conf
-rw-r--r-- 1 root root  347 Feb 15 04:04 iwlwifi.conf
-rw-r--r-- 1 root root  379 Jan  7  2022 mdadm.conf
-rw-r--r-- 1 root root  117 Mar 27 06:29 nvidia-graphics-drivers-kms.conf
-rw-r--r-- 1 root root 2739 May 31 08:44 sof-dyndbg.conf

inxi -SMA
System:
  Host: XPS17-9730 Kernel: 6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f
    arch: x86_64 bits: 64 Desktop: GNOME v: 44.0 Distro: Ubuntu 23.04 (Lunar
    Lobster)
Machine:
  Type: Laptop System: Dell product: XPS 17 9730 v: N/A
    serial: <superuser required>
  Mobo: Dell model: 01Y5VN v: A00 serial: <superuser required> UEFI: Dell
    v: 1.2.2 date: 03/23/2023
Audio:
  Device-1: Intel driver: sof-audio-pci-intel-tgl
  Device-2: Creative Pebble Pro type: USB
    driver: hid-generic,snd-usb-audio,usbhid
  Sound API: ALSA v: k6.4.0-rc1-pr4381-5160-default-g812ee1d6cb3f
    running: yes
  Sound Server-1: PipeWire v: 0.3.65 running: yes

**Comment by plbossart** [comment]
It's very different @KarthikBala2003, now the error is this
````
[    8.446375] sof-audio-pci-intel-tgl 0000:00:1f.3: Direct firmware load for intel/sof-tplg/sof-rpl-rt711-l0-rt1316-l12-rt714-l3.tplg failed with error -2
````

That means your hardware was perfectly detected but you are missing a configuration file which I provided earlier in this comment: https://github.com/thesofproject/linux/issues/4380#issuecomment-1563496427

**Comment by KarthikBala2003** [comment]
Got it. I missed this step. Thanks so much. Try again and keep you posted

**Comment by KarthikBala2003** [comment]
Hello Mr. Plbossart,

Thanks so much for your help and really appreciated your support.

I followed the instruction you gave me (copied the config file under lib folder).  
Finally sound is working great.  One problem though Bluetooth is not detected and the option is greyed out, I can't enable it. 
Can we fix this issue (see the attached screenshot)?

![BT 2023-05-31 10-24-19](https://github.com/thesofproject/linux/assets/12576200/4ff368e1-1097-4fcb-9b3d-74d142817dce)

[alsa-info.txt](https://github.com/thesofproject/linux/files/11616045/alsa-info.txt)
[dmesg.txt](https://github.com/thesofproject/linux/files/11616047/dmesg.txt)
[dsdt.zip](https://github.com/thesofproject/linux/files/11616048/dsdt.zip)

**Comment by KarthikBala2003** [comment]
![SoundWire 2023-05-31 10-22-58](https://github.com/thesofproject/linux/assets/12576200/4ee207f2-3555-4311-abb9-3c9bf9cde178)

**Comment by plbossart** [comment]
Good to hear, we will submit the changes for 6.5.

The Bluetooth problem is likely a kernel configuration, for our tests we prune the configs to a bare minimum to save on compilation time. Either you try and compile yourself, or wait a couple of weeks (probably more like 8 weeks) for everything to show up in the linux stable kernel.

**Comment by KarthikBala2003** [comment]
Superb. Thanks again. How do I know which kernel stable release includes this fix?

Just curious, my own development and Linux custom kernel build process is there any document available? 
You helped me a lot but I just followed your instruction to fix the issue, my questions are

1. Where can I my custom config files related to Wif/Bluetooth or any other device drivers?
       -  . Is this the location /lib/firmware/intel/sof-tplg/?
2.. Where do all custom config unit files go? Is it placed under /etc/modprobe.d?
3. After making all config/firmware changes, how can I compile a custom kernel? 
    - You already helped my with this and created .deb package, for my learning can I build the package and created deb file. 
    -   What steps do I need to follow?  Is there set of tools/packages need install to create .deb files?
    
I would appreciate it if you help me to do it myself.

**Comment by plbossart** [comment]
You don't really need a custom config, what you want to use is the default config used by your Ubuntu release (usually found in /boot/config-something) and use it as a .config where your kernel code is. The configuration is used only for compilation and does not need to be installed.

You can look at the example for GitHub actions:

````
gcc-sof-x86_64:
    name: "GCC build X86_64"
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@v3

      - name: Fetch kconfig
        uses: actions/checkout@v3
        with:
          repository: 'thesofproject/kconfig'
          path: 'kconfig'

      - name: Install libelf
        run: |
          sudo apt update
          sudo apt install -y libelf-dev

      - name: Install debhelper
        run: |
          sudo apt update
          sudo apt install -y debhelper

      - name: build start
        run: |
          export ARCH=x86_64 KCFLAGS="-Wall -Werror"
          export MAKEFLAGS=j"$(nproc)"
          bash kconfig/kconfig-sof-default.sh <<< this would be replaced by cp /boot/config-something .config
          make
          make bindeb-pkg
````

**Comment by KarthikBala2003** [comment]
Great. I will check this one.
Thanks again.
Can I get a notification email with stable release number included this fix?

**Comment by plbossart** [comment]
> Can I get a notification email with stable release number included this fix?

sorry, that service does not exist.

**Comment by KarthikBala2003** [comment]
No worries. I will check next few stable releases.
Thanks again for your help.

**Comment by KarthikBala2003** [comment]
I just want to check with you guys any tentative release date for these changes? 
I have installed 6.4.0-060400rc6-generic
I didn't find these changes

**Comment by plbossart** [comment]
this is queued for 6.5 and may be picked up earlier in a 6.4.x stable release. We have no say on which commits get picked when.

**Comment by KarthikBala2003** [comment]
Thanks for the update

**Comment by KarthikBala2003** [comment]
Can someone confirm this fix was added into 6.5-rc1?

**Comment by marcmerlin** [comment]
I may have a similar issue, sound works with ubuntu 6.5.0-14 kernel in ubuntu 23, but not in debian12 with the same kernel
https://github.com/thesofproject/linux/issues/4758

No sound on speakers either with 6.6.8, but everything looks like it's setup right, all channels are umuted, I see sound playing, just no output on speakers, although it works if I plug headphones

EDIT: in my case alsa-ucm-conf was required and missing
